### PR TITLE
Github actions with three different JDKs and OSes.

### PR DESCRIPTION
**Pull request by bmarwell**

### CODE REVIEW

**Comment by kriegaex** [comment]
Thanks, this was also on my to-do list, getting rid of the obsolete Travis stuff and setting up GitHub actions instead. Did you verify those builds actually work on your own fork? I just see [this run](https://github.com/bmarwell/aspectj-maven-plugin/actions/runs/881179414), and there seems to be a problem.

BTW, I noticed build problems on JDK 16, but am quite busy with my daytime job and did not look into it yet. We are going to get all of it sorted out, and your build configs could be an important part of pointing out possible problems and incompatibilities. I just was not able to focus on it yet, I just wanted to get the milestone out and ran all tests locally on my Windows box.

**Comment by bmarwell** [comment]
Thanks, I did not pay attention to this because this was mostly copied from my Apache Shiro Branch. But I added some invalid cache exclusions, which is why it failed.

New build will at least run: https://github.com/bmarwell/aspectj-maven-plugin/actions/runs/881364653

It does not go green, because it is not portable. Fixing this should be one of our next goals. I will open an issue for this.

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            maven-${{ matrix.os }}-
+
+      - name: Set up JDK
+        uses: actions/setup-java@v2.1.0
+        with:
+          java-version: ${{ matrix.java.version }}
+          distribution: ${{ matrix.java.dist }}
+
+      - name: Build with Maven
+        run: mvn --show-version --errors --batch-mode clean install site
```
Comment (bmarwell) [inline .github/workflows/maven.yml:91]: I think, instead of `install` we might want to use `verify`.

Comment (bmarwell) [reply .github/workflows/maven.yml:91]: Also, `-Dinvoker.streamLogs=true` or `-Dinvoker.streamLogsOnFailures=true`. Your choice @kriegaex

**Review by kriegaex — COMMENTED** [review]
As for the build matrix, actually I am more concerned about the plugin working on multiple platforms and Java versions than it being buildable on all those platforms. For the last few years, there was never more than one developer working on it, and not often or much either. I am not saying that some things should not be addressed, e.g. why the build fails on Java 11 because the Maven Site plugin seems to look for tools.jar. So that one obviously needs an update. Maven Central also revealed some vulnerabilities in some dependencies in a [report](https://sbom.lift.sonatype.com/report/T1-0ff0976f7f21c391f20f-dfa463bcb34dd-1622009500-fb5c1335856646e2ad417879bb4cf6bc) generated automatically. So some stuff needs updating anyway or some dependencies need managing.

Anyway, the main concern for me is to run the integration tests on different platforms, Java versions and maybe JVM types (even though there I am expecting different results rather with regard to AspectJ itself than AspectJ Maven).

I guess, having a CI build at all is such an important thing that I will probably merge this anyway and then maybe modify it later. But if you have ideas how to improve it before the merge, I would welcome that too. I am not trying to block anything, because I hate slow OSS review processes myself. But all your PRs have just been created today. I just want to discuss some things. I do appreciate your speed and enthusiasm, though. I am happy not to be alone in this.

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
```
[the remark this answers is not in the document]
Comment (bmarwell) [reply .github/workflows/maven.yml:91]: After the matrix is done, we can start another build job using docker+graphviz which deploys.
tbh I am not a fan of deploying from GH actions, at least not for releases. But adding a second build chain right after with only one JDK and graphviz will solve this.
Can do this in another PR.

**Comment by bmarwell** [comment]
> Anyway, the main concern for me is to run the integration tests on different platforms, Java versions and maybe JVM types (even though there I am expecting different results rather with regard to AspectJ itself than AspectJ Maven).

Well, then this PR is just right for you! While they should (mostly) generate the same output due to `maven.compiler.target`, the ITs are being run on different platforms and JDKs as well.

**Comment by kriegaex** [comment]
Well, yes, but there are several issues of things failing or passing on different Java versions. Both Javadoc and PlantUML behave differently and throw different errors, for example, and the site looks different. I am in the process of sorting that out. For PlantUML and Javadoc I have the commits in place locally, but I am still investigating an issue about project reports no longer being integrated into the Maven site after upgrading the parent Mojohaus POM and a few plugins and dependencies due to CVS reports. I am calling it a day, 20:15 here already, and I am getting hungry.

**Comment by bmarwell** [comment]
> Well, yes, but there are several issues of things failing or passing on different Java versions. Both Javadoc and PlantUML behave differently and throw different errors, for example, and the site looks different.

That is a check job only not for deployment. For deployment, I propose another job as already mentioned, which is being executed after this one.

File: .github/workflows/maven.yml (hunk 1 above)
Comment (kriegaex) [reply .github/workflows/maven.yml]: I used `verify` and `-Dinvoker.streamLogsOnFailures=true`, according to your suggestions.

File: .github/workflows/maven.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kriegaex) [reply .github/workflows/maven.yml]: For now I have experimentally added a GH action for setting up Graphviz to your PR in https://github.com/dev-aspectj/aspectj-maven-plugin/pull/2/commits/43c5afcff946b85776c9de38415e69aca7b58888. I am going to merge now, then we can see how the builds behave and decide how to proceed further.

**Comment by bmarwell** [comment]
happy to help.

**Pull request merged**